src/index.js

```diff
@@ -83,49 +83,50 @@ export class Scene extends React.Component {
  * {primitive: box; width: 10} to 'primitive: box; width: 10'
  */
 export function serializeComponents (props) {
-  var components = AFRAME.components;
+  const components = AFRAME.components;
+  const serialProps = {};
 
-  let serialProps = {};
   Object.keys(props).forEach(component => {
+    const value = props[component];
+    const { constructor } = value;
+    const definition = components[component];
+    const schema = definition ? definition.schema : {};
```

Comment:
Need to safely handle accessing the schema when it does not exist (so any property that's not a component)